packages/go_router_builder/lib/src/path_utils.dart

```diff
@@ -0,0 +1,160 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'route.dart';
+
+final RegExp _parameterRegExp = RegExp(r':(\w+)(\((?:\\.|[^\\()])+\))?');
+
+/// Converts a [pattern] such as `/user/:id` into [RegExp].
+///
+/// The path parameters can be specified by prefixing them with `:`. The
+/// `parameters` are used for storing path parameter names.
+///
+///
+/// For example:
+///
+///  `pattern` = `/user/:id/book/:bookId`
+///
+///  The `parameters` would contain `['id', 'bookId']` as a result of calling
+///  this method.
+///
+/// To extract the path parameter values from a [RegExpMatch], pass the
+/// [RegExpMatch] into [extractPathParameters] with the `parameters` that are
+/// used for generating the [RegExp].
+RegExp patternToRegExp(String pattern, List<String> parameters) {
+  final StringBuffer buffer = StringBuffer('^');
```

Comment:
This file is this commit ([chore: Copy path_utils.dart from go_router](https://github.com/flutter/packages/pull/4524/commits/c9487b301293bedc308e1b4561055db1f9ec6e61)) a literal copy past from `go_router`'s `path_utils.dart` file https://github.com/flutter/packages/blob/main/packages/go_router/lib/src/path_utils.dart